### PR TITLE
refactor(examples): Migrate `quickstart-monai` to Message API

### PR DESCRIPTION
**Pull request by psfoley**
This PR:
- Updates the example to use the Message API
- Updates MONAI to the latest version of 

Changes to data downloading and loading (from Huggingface) will be left for a future PR

### CODE REVIEW

File: examples/quickstart-monai/monaiexample/server_app.py [hunk 1]
```diff
@@ -2,45 +2,41 @@
...
+    """Main entry point for the ServerApp."""
 
+    # Read run config
+    fraction_train: float = context.run_config["fraction-train"]
+    num_rounds: int = context.run_config["num-server-rounds"]
 
-def server_fn(context: Context):
-
-    # Init model
+    # Load global model
```
Comment (panh99) [inline examples/quickstart-monai/monaiexample/server_app.py:22]: ~~It seems that we are not loading any pre-trained model weights here. Would it be better to say:~~
```suggestion
    # Initialize the global model
```

Nvm, I noticed that we use the same comment in all quickstart examples.

File: examples/quickstart-monai/monaiexample/client_app.py [hunk 2]
```diff
@@ -1,41 +1,74 @@
...
-from flwr.common import Context
+from flwr.app import ArrayRecord, Context, Message, MetricRecord, RecordDict
+from flwr.clientapp import ClientApp
 
-from monaiexample.task import get_params, load_data, load_model, set_params, test, train
+from monaiexample.task import (
+    get_params,
+    load_data,
+    load_model,
+    set_params,
```
Comment (jafermarq) [inline examples/quickstart-monai/monaiexample/client_app.py]:
```suggestion
    load_data,
    load_model,
```
We are now using the built-in mechanism to convert torch state_dicts to `ArrayRecord`. We can remove those utility functions in `task.py`

File: examples/quickstart-monai/monaiexample/server_app.py [hunk 3]
```diff
@@ -2,45 +2,41 @@
 
 from typing import List, Tuple
 
-from flwr.common import Context, Metrics, ndarrays_to_parameters
-from flwr.server import ServerApp, ServerAppComponents, ServerConfig
-from flwr.server.strategy import FedAvg
+import torch
+from flwr.app import ArrayRecord, ConfigRecord, Context, MetricRecord
```
Comment (jafermarq) [inline examples/quickstart-monai/monaiexample/server_app.py]:
```suggestion
from flwr.app import ArrayRecord, Context
```

File: examples/quickstart-monai/monaiexample/server_app.py [hunk 4]
```diff
@@ -2,45 +2,41 @@
 
 from typing import List, Tuple
```
Comment (jafermarq) [inline examples/quickstart-monai/monaiexample/server_app.py]:
```suggestion
```

File: examples/quickstart-monai/monaiexample/server_app.py [hunk 5]
```diff
@@ -2,45 +2,41 @@
...
 
-from flwr.common import Context, Metrics, ndarrays_to_parameters
-from flwr.server import ServerApp, ServerAppComponents, ServerConfig
-from flwr.server.strategy import FedAvg
+import torch
+from flwr.app import ArrayRecord, ConfigRecord, Context, MetricRecord
+from flwr.serverapp import Grid, ServerApp
+from flwr.serverapp.strategy import FedAvg
 
 from monaiexample.task import get_params, load_model
```
Comment (jafermarq) [inline examples/quickstart-monai/monaiexample/server_app.py]:
```suggestion
from monaiexample.task import load_model
```

File: examples/quickstart-monai/monaiexample/task.py [hunk 6]
```diff
@@ -40,21 +40,26 @@ def set_params(model, ndarrays):
...
+            loss = loss_function(model(images.to(device)), labels.to(device))
+            loss.backward()
             optimizer.step()
+            running_loss += loss.item()
+    avg_trainloss = running_loss / len(train_loader)
+    return avg_trainloss
 
 
-def test(model, test_loader, device):
+def test_func(model, test_loader, device):
```
Comment (panh99) [inline examples/quickstart-monai/monaiexample/task.py:62]: Should we compute the average loss here? @jafermarq @psfoley

Comment (jafermarq) [reply examples/quickstart-monai/monaiexample/task.py:62]: it sounds reasonable to me. We an accumulate the lost and then compute the average as in the `quickstart-pytorch` example: https://github.com/adap/flower/blob/fab093532cf6f8623818e73246bb15da183a8a79/examples/quickstart-pytorch/pytorchexample/task.py#L108

**Review by panh99 — APPROVED** [review]
LGTM!
